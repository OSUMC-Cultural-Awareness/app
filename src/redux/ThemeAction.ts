import { Store } from ".";

export const updateTheme = (
  theme: Store["theme"]
): { type: string; payload: Store["theme"] } => ({
  type: "UPDATE_THEME",
  payload: theme,
});
