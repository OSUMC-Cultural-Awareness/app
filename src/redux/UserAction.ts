import { Store } from ".";

export const updateUser = (
  user: Store["user"]
): { type: string; payload: Store["user"] } => ({
  type: "UPDATE_USER",
  payload: user,
});

export const resetUser = (): {
  type: string;
  payload: Record<string, unknown>;
} => ({
  type: "RESET_USER",
  payload: {},
});
