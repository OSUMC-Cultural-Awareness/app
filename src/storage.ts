// Locations to store information in {@link AsyncStorage}.
export namespace Storage {
  // Location to store a user's email locally on user's device.
  export const RememberedEmail = "@rememberedEmail";

  // Location to store theme "Light" or "Dark" on user's device.
  export const Theme = "@theme";

  // Location to store ledger on user's device
  export const Ledger = "@ledger";
}
