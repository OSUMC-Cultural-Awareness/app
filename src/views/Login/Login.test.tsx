/**
 * @jest-environment jsdom
 */
import React from "react";
import "react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { shallow, mount } from "enzyme";
import AsyncStorageMock from "@react-native-async-storage/async-storage/jest/async-storage-mock";
import AsyncStorage from "@react-native-async-storage/async-storage";

import Login from ".";
import { USER_INITIAL_STATE, THEME_INITIAL_STATE } from "../../redux";
import { Storage } from "../../storage";
import { Routes } from "../../routes";

jest.useFakeTimers();

const LoginTest = () => {
  const Stack = createStackNavigator<Routes>();

  const mockStore = configureStore([]);

  return (
    <Provider
      store={mockStore({
        user: USER_INITIAL_STATE,
        theme: THEME_INITIAL_STATE,
      })}
    >
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Login">
          <Stack.Screen name="Login" component={Login} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
};

describe("Login", () => {
  test("Renders", () => {
    expect(shallow(<LoginTest />)).toMatchSnapshot();
  });

  test("Remember Me Default to Unchecked", () => {
    const page = mount(<LoginTest />);
    const checkbox = page.find("Checkbox").first();
    expect(checkbox.props().status).toBe("unchecked");
  });

  test("Remember Me Checked When Email Set in AsyncStorage", async () => {
    AsyncStorageMock.getItem = jest.fn().mockReturnValue("test@gmail.com");
    const page = mount(<LoginTest />);

    setImmediate(() => {
      page.update();

      const checkbox = page.find("Checkbox").first();
      expect(AsyncStorage.getItem).toBeCalledWith(Storage.RememberedEmail);
      expect(checkbox.props().status).toBe("checked");
    });
  });
});
