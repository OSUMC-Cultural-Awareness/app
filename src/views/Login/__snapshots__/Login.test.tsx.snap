// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ForwardRef(NavigationContainer)>
    <StackNavigator
      initialRouteName="Login"
    >
      <Screen
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Login)",
            "type": [Function],
          }
        }
        name="Login"
      />
    </StackNavigator>
  </ForwardRef(NavigationContainer)>
</Provider>
`;
